axelor-account/src/main/java/com/axelor/apps/account/service/invoice/InvoiceLineServiceImpl.java

```diff
@@ -740,4 +755,24 @@ public Map<String, Map<String, Object>> setScale(InvoiceLine invoiceLine, Invoic
 
     return attrsMap;
   }
+
+  @Override
+  public String computeInvoiceLineTaxLineSetDomain(int operationTypeSelect) {
+    LocalDate todayDate = LocalDate.now();
+    String domain = "(self.endDate > :date OR self.endDate IS NULL) ";
+    if (operationTypeSelect == 3 || operationTypeSelect == 4) {
```

Comment:
Here it is better to use repository variables, so instead of 3 and 4 you should use InvoiceRepository.OPERATION_TYPE_CLIENT_SALE and InvoiceRepository.OPERATION_TYPE_CLIENT_REFUND